File: automated-test-orchestrator-api/src/domain/mapping.ts
// src/domain/mapping.ts

export interface Mapping {
  id: string;
  mainComponentId: string;
  mainComponentName?: string;
  testComponentId: string;
  testComponentName?: string;
  isDeployed?: boolean;
  isPackaged?: boolean;
  createdAt: Date;
  updatedAt: Date;
}

export type CreateMappingDTO = Omit<Mapping, 'id' | 'createdAt' | 'updatedAt'>;
export type UpdateMappingDTO = Partial<Omit<Mapping, 'id' | 'mainComponentId' | 'createdAt' | 'updatedAt'>>;

File: testify-cli/internal/model/mapping.go
// automated-test-orchestrator-cli/internal/model/mapping.go
package model

import "time"

// CliMapping represents a single mapping record returned by the API.
type CliMapping struct {
	ID                string    `json:"id"`
	MainComponentID   string    `json:"mainComponentId"`
	MainComponentName *string   `json:"mainComponentName,omitempty"`
	TestComponentID   string    `json:"testComponentId"`
	TestComponentName *string   `json:"testComponentName,omitempty"`
	IsDeployed        *bool     `json:"isDeployed,omitempty"`
	IsPackaged        *bool     `json:"isPackaged,omitempty"`
	CreatedAt         time.Time `json:"createdAt"`
	UpdatedAt         time.Time `json:"updatedAt"`
}

// CreateMappingRequest is the structure for the POST /mappings request body.
type CreateMappingRequest struct {
	MainComponentID   string  `json:"mainComponentId"`
	MainComponentName *string `json:"mainComponentName,omitempty"`
	TestComponentID   string  `json:"testComponentId"`
	TestComponentName *string `json:"testComponentName,omitempty"`
	IsDeployed        *bool   `json:"isDeployed,omitempty"`
	IsPackaged        *bool   `json:"isPackaged,omitempty"`
}
